src/controller/InvokeInteraction.h

```diff
@@ -81,6 +83,15 @@ InvokeCommandRequest(Messaging::ExchangeManager * aExchangeMgr, const SessionHan
     ReturnErrorOnFailure(commandSender->AddRequestData(commandPath, requestCommandData, timedInvokeTimeoutMs));
     ReturnErrorOnFailure(commandSender->SendCommandRequest(sessionHandle, responseTimeout));
 
+    // If requested by the caller, provide a way to cancel the invoke interaction.
+    if (outCancelFn != nullptr)
+    {
+        *outCancelFn = [rawDecoderPtr = decoder.get(), rawCommandSender = commandSender.get()]() {
+            chip::Platform::Delete(rawCommandSender);
+            chip::Platform::Delete(rawDecoderPtr);
```

Comment:
So what happens if someone calls this cancel function after OnDone?

Reply:
Bad things! This is intended to be just for internal use by the device commissioner at this point, but I don't know if there is a good way to enforce that without moving a bunch of code around.

I think medium term `TypedCommandCallback` should maybe go away and we should instead have something more like `ReadClient` (or `CommandSender`) where the client can explicitly manage the object lifetime, and maybe a `Auto...` variant / wrapper that heap-allocates and frees automatically?

Reply:
I've documented `InvokeCancelFn` as Internal, so we don't have to consider it API and retain support for this exact mechanism going forward.